[inductor] generate nan in the cpp backend

desertfire opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #89289

Summary: Fixes https://github.com/pytorch/torchdynamo/issues/1797

cc @mlazos @soumith @voznesenskym @yanboliang @penguinwu @anijain2305 @EikanWang @jgong5 @Guobing-Chen @chunyuan-w @XiaobingSuper @zhuhaozhe @blzheng @Xia-Weiwen @wenzhe-nrv @jiayisunx @peterbell10

desertfire committed:
[inductor] generate nan in the cpp backend

Summary: Fixes https://github.com/pytorch/torchdynamo/issues/1797

[ghstack-poisoned]

desertfire committed:
Update on "[inductor] generate nan in the cpp backend"


Summary: Fixes https://github.com/pytorch/torchdynamo/issues/1797

cc mlazos soumith voznesenskym yanboliang penguinwu anijain2305 EikanWang jgong5 Guobing-Chen chunyuan-w XiaobingSuper zhuhaozhe blzheng Xia-Weiwen wenzhe-nrv jiayisunx peterbell10

[ghstack-poisoned]

desertfire commented:
@pytorchbot merge

jgong5 commented on torch/_inductor/codegen/cpp.py:
```diff
@@ -268,6 +269,8 @@ def constant(val, dtype):
         elif val == float("-inf"):
             quote = f"-std::numeric_limits<{DTYPE_TO_CPP[dtype]}>::infinity()"
+        elif math.isnan(val):
+            return f"std::numeric_limits<{DTYPE_TO_CPP[dtype]}>::quiet_NaN()"
```
```suggestion
            quote = f"std::numeric_limits<{DTYPE_TO_CPP[dtype]}>::quiet_NaN()"
```
Please add the case to cover it too. Code sequence like below can trigger this vec path:
```python
    a = torch.tensor(float("nan"))
    b = torch.empty([100])
    b.fill_(a)
```

desertfire committed:
Update on "[inductor] generate nan in the cpp backend"


Summary: Fixes https://github.com/pytorch/torchdynamo/issues/1797

cc mlazos soumith voznesenskym yanboliang penguinwu anijain2305 EikanWang jgong5 Guobing-Chen chunyuan-w XiaobingSuper zhuhaozhe blzheng Xia-Weiwen wenzhe-nrv jiayisunx peterbell10

[ghstack-poisoned]

desertfire committed:
Update on "[inductor] generate nan in the cpp backend"


Summary: Fixes https://github.com/pytorch/torchdynamo/issues/1797

cc mlazos soumith voznesenskym yanboliang penguinwu anijain2305 EikanWang jgong5 Guobing-Chen chunyuan-w XiaobingSuper zhuhaozhe blzheng Xia-Weiwen wenzhe-nrv jiayisunx peterbell10

[ghstack-poisoned]

desertfire committed:
Update on "[inductor] generate nan in the cpp backend"


Summary: Fixes https://github.com/pytorch/torchdynamo/issues/1797

cc mlazos soumith voznesenskym yanboliang penguinwu anijain2305 EikanWang jgong5 Guobing-Chen chunyuan-w XiaobingSuper zhuhaozhe blzheng Xia-Weiwen wenzhe-nrv jiayisunx peterbell10

[ghstack-poisoned]

desertfire commented:
@pytorchbot merge
